### PR TITLE
Replace scale.useRawDomain with scale.domain = ‘unaggregated’

### PR DESCRIPTION
**Pull request by domoritz**
Part of #1602

### CODE REVIEW

File: src/compile/scale/domain.ts [hunk 1]
```diff
@@ -143,7 +143,7 @@ export function domainSort(model: Model, channel: Channel, scaleType: ScaleType)
 function _useRawDomain (scale: Scale, model: Model, channel: Channel) {
   const fieldDef = model.fieldDef(channel);
 
-  return scale.useRawDomain && //  if useRawDomain is enabled
+  return scale.domain === 'pre-aggregation' && //  if useRawDomain is enabled
```
Comment (kanitw) [inline src/compile/scale/domain.ts]: // if use pre-aggregate domain.

File: src/compile/scale/domain.ts [hunk 2]
```diff
@@ -15,7 +15,7 @@ import {Model} from '../model';
 export default function domain(scale: Scale, model: Model, channel:Channel): any[] | VgDataRef | FieldRefUnionDomain {
   const fieldDef = model.fieldDef(channel);
 
-  if (scale.domain) { // explicit value
+  if (scale.domain && scale.domain !== 'pre-aggregation') { // explicit value
```
Comment (kanitw) [inline src/compile/scale/domain.ts]: is `"pre-aggregate"` better?  (It's at least shorter.)

Comment (domoritz) [inline src/compile/scale/domain.ts]: Sure. Will replace.

File: src/scale.ts [hunk 3]
```diff
@@ -170,8 +163,11 @@ export interface Scale {
   type?: ScaleType;
   /**
    * The domain of the scale, representing the set of data values. For quantitative data, this can take the form of a two-element array with minimum and maximum values. For ordinal/categorical data, this may be an array of valid input values.
+   *
+   * If the domain is 'pre-aggregation', we use the source data range as scale domain instead of aggregated data for aggregate axis.
+   * This property only works with aggregate functions that produce values within the raw data domain (`"mean"`, `"average"`, `"stdev"`, `"stdevp"`, `"median"`, `"q1"`, `"q3"`, `"min"`, `"max"`). For other aggregations that produce values outside of the raw data domain (e.g. `"count"`, `"sum"`), this property is ignored.
```
Comment (kanitw) [inline src/scale.ts]: I think stdev and stdevp can be out of min/max range.

File: test/compile/scale/domain.test.ts [hunk 4]
```diff
@@ -145,14 +144,14 @@ describe('compile/scale', () => {
           assert.deepEqual(_domain.data, SOURCE);
         });
 
-      it('should return the raw domain if useRawDomain is true for year T',
+      it('should return the pre-aggregationd domain if requested for year T',
```
Comment (kanitw) [inline test/compile/scale/domain.test.ts]: typo: extra "d"

File: src/compile/scale/domain.ts (hunk 2 above)
Comment (domoritz) [inline src/compile/scale/domain.ts]: I like `pre-aggregation` better because it doesn't sound like a verb.

File: src/scale.ts (hunk 3 above)
Comment (domoritz) [inline src/scale.ts]: Yes.

File: test/compile/scale/domain.test.ts (hunk 4 above)
Comment (domoritz) [inline test/compile/scale/domain.test.ts]: global replace issue

File: src/scale.ts (hunk 3 above)
Comment (kanitw) [inline src/scale.ts]: Yep, and we have removed two of these from `SHARED_DOMAIN_OPS` already, so we just need to update comment here :)

File: src/compile/scale/domain.ts (hunk 1 above)
Comment (domoritz) [inline src/compile/scale/domain.ts]: Fixed

File: src/scale.ts (hunk 3 above)
Comment (domoritz) [inline src/scale.ts]: Done

File: src/compile/scale/domain.ts (hunk 2 above)
Comment (kanitw) [inline src/compile/scale/domain.ts]: "pre-aggregate"

**Comment by domoritz** [comment]
@kanitw Note that I removed the config property. If you need it for compass, shout.

**Comment by kanitw** [comment]
Yes, I do.

File: src/compile/scale/domain.ts (hunk 2 above)
Comment (domoritz) [inline src/compile/scale/domain.ts]: could still be a verb

**Comment by domoritz** [comment]
Done.

File: src/compile/scale/domain.ts (hunk 2 above)
Comment (arvind) [inline src/compile/scale/domain.ts]: "unaggregated"?

File: src/compile/scale/domain.ts [hunk 5]
```diff
@@ -51,12 +51,12 @@ export default function domain(scale: Scale, model: Model, channel:Channel): any
     };
   }
 
-  // FIXME refactor _useRawDomain's signature
-  const useRawDomain = _useRawDomain(scale, model, channel);
+  // FIXME refactor signature
+  const noAggDomain = _useUnaggregatedDomain(scale, model, channel);
```
Comment (kanitw) [inline src/compile/scale/domain.ts]: remove _ for the method?

Comment (domoritz) [inline src/compile/scale/domain.ts]: Done. Merge when you feel like it.

**Pull request merged**